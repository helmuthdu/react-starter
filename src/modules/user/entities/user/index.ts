export * from './user.type.ts';
export * from './user.model';
