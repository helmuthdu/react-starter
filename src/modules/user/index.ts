export { routes } from './routes';
export * from './stores';
