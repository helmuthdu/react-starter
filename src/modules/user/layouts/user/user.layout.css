@import 'https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro';

body,
html {
  overflow: hidden auto;
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  background-color: #fff;
  font-size: 16px;
}
