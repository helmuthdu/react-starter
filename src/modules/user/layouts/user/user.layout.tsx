import React, { Fragment } from 'react';
import { Outlet } from 'react-router-dom';
import './user.layout.scss';

export const UserLayout = () => (
  <Fragment>
    <Outlet />
  </Fragment>
);

export default UserLayout;
