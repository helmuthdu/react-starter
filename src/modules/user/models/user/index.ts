export * from './user.scheme';
export * from './user.model';
