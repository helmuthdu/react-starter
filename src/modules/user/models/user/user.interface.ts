export interface IUser {
  name: string;
  username: string;
  email: string;
  isLogged: boolean;
  token: string;
}
