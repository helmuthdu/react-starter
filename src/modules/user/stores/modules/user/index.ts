export * from './actions';
export * from './selectors';
export * from './reducer';
export * from './state';
export * from './types';

export const name = 'user';
