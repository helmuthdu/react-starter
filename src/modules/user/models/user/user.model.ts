import { UserScheme } from './user.scheme';

export class User implements UserScheme {
  name: '';
  username: '';
  email: '';
  isLogged: false;
  token: '';
}
