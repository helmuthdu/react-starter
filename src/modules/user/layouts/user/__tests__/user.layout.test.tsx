import { render } from '@testing-library/react';
import React from 'react';
import { UserLayout } from '../user.layout';

describe('user/layouts -> DefaultLayout component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<UserLayout />);
    expect(asFragment()).toBeDefined();
  });
});
