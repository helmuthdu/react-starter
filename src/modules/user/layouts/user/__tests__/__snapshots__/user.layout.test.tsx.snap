// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user/layouts -> DefaultLayout component should match snapshot 1`] = `
<DocumentFragment>
  <p>
    lorem ipsum
  </p>
</DocumentFragment>
`;

exports[`user/layouts -> DefaultLayoutRoute component should match snapshot 1`] = `
<DocumentFragment>
  <p>
    lorem ipsum
  </p>
</DocumentFragment>
`;
